project_parent_task_filter/models/project.py

```diff
@@ -0,0 +1,29 @@
+from odoo import api, models
+
+
+class TaskSub(models.Model):
+    _inherit = "project.task"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super(TaskSub, self).create(vals_list)
+        for vals in vals_list:
+            if vals.get("parent_id", False):
```

Comment:
It might be a good idea to add an init hook to ensure that the historical data remains consistent.